lib/bing_translator.rb

```diff
@@ -88,11 +92,14 @@ def request_new_access_token
       http.verify_mode = OpenSSL::SSL::VERIFY_NONE if @skip_ssl_verify
 
       response = http.post(COGNITIVE_ACCESS_TOKEN_URI.path, '', headers)
-      if response.code != '200'
-        raise Exception.new("Unsuccessful Access Token call: Code: #{response.code} (Invalid credentials?)")
-      else
+      case response
+      when Net::HTTPSuccess
         @access_token_expiration_time = Time.now + 480
         response.body
+      when Net::HTTPServerError
+        raise UnavailableException.new("#{response.code}: Credentials server unavailable")
+      else
+        raise Exception.new("Unsuccessful Access Token call: Code: #{response.code} (Invalid credentials?)")
```

Comment:
I believe the specialized error class got lost here when merging master.

The new spec seems to expect `AuthenticationException`

Reply:
Sorry for the sloppy merge.  I've fixed this in 1f71d42 and added tests to ensure the correct exception is being raised.  Thanks!